build.gradle

```diff
@@ -533,16 +531,12 @@ allprojects {
 wrapper {
   distributionType = 'ALL'
   doLast {
-    final DistributionLocator locator = new DistributionLocator()
```

Comment:
@dbwiddis I have a few comments and concerns:
 - the `org.gradle.util.DistributionLocator` seems to be replaced with `org.gradle.util.internal.DistributionLocator` which we should be using
 - by trusting the `distributionUrl` we basically allow anyone to supply own distribution with whatever SHAs it could have, the `DistributionLocator` ensures that SHAs are always coming from the trusted source

Reply:
I would think we should not be using a package marked `internal`.

Note that I sourced the changes from https://github.com/aiven/s3-connector-for-apache-kafka/pull/223 (and confirmed the code).

Reply:
> I would think we should not be using a package marked internal.

That would be ideal, I agree, so we could leverage the fact wrapper task has locator which is used in `getDistributionUrl()` method to do exactly the same thing:

```
final URI sha256Uri = new URI(getDistributionUrl() + ".sha256")
``` 

> Note that I sourced the changes from https://github.com/aiven/s3-connector-for-apache-kafka/pull/223 (and confirmed the code).

:-) I would not replicate this approach :)

Reply:
I updated the PR to use `getDistributionUrl` but the locator code is only called if the `distributionUrl` is null, so it doesn't seem to answer your concern...
```java
public String getDistributionUrl() {
    if (distributionUrl != null) {
        return distributionUrl;
    } else if (gradleVersionResolver.getGradleVersion() != null) {
        return locator.getDistributionFor(gradleVersionResolver.getGradleVersion(), distributionType.name().toLowerCase(Locale.ENGLISH)).toString();
    } else {
        return null;
    }
}
```

Reply:
Digging into the code again, I think the `getDistributionUrl()` is an improvement, but I don't see the security concern with specifying a custom URL
 - It looks like an intentional design to allow companies to host their own internal repo
 - the code in the locator (both old and new locations) eventually [breaks down to this](https://github.com/gradle/gradle/blob/b75cbf987d7c9c76b64e274f5997e2769862091b/subprojects/core/src/main/java/org/gradle/util/DistributionLocator.java#LL65C13-L65C127), taking `repoistoryUrl` as an input, which a user can specify.
```java
new URI(repositoryUrl + "/" + archiveName + "-" + version.getVersion() + "-" + archiveClassifier + ".zip");
```
  - All it is really doing is appending a sha256 to the properties file, which a user could do manually anyway if they wanted.  I could just replace the last line here:
```
➜  OpenSearch git:(distribution-locator) cat gradle/wrapper/gradle-wrapper.properties
distributionBase=GRADLE_USER_HOME
distributionPath=wrapper/dists
distributionUrl=https\://services.gradle.org/distributions/gradle-7.6-all.zip
networkTimeout=10000
zipStoreBase=GRADLE_USER_HOME
zipStorePath=wrapper/dists
distributionSha256Sum=312eb12875e1747e05c2f81a4789902d7e4ec5defbd1eefeaccc08acf096505d
```

Reply:
May be I am over complicating things :) thanks @dbwiddis